frontend/apps/crates/entry/module/tapping-board/play/src/base/game/playing/actions.rs

```diff
@@ -2,19 +2,27 @@ use super::state::*;
 use components::module::_common::play::prelude::*;
 use shared::domain::module::body::tapping_board::Next;
 use std::rc::Rc;
+use utils::unwrap::UnwrapJiExt;
 
 impl PlayState {
     pub fn select(state: Rc<Self>, index: usize) {
+        // Kill playback for all already selected traces
+        state
+            .selected_set
+            .lock_ref()
+            .iter()
+            .for_each(|index| state.traces.get(*index).unwrap_ji().kill_playback());
+
```

Comment:
Small optimization: Instead of iterating over every trace, only end playback of traces which are in the `current_set`